lib/appsignal/check_in.rb

```diff
@@ -84,14 +87,26 @@ def heartbeat(identifier, continuous: false)
 
       # @api private
       def transmitter
-        @transmitter ||= Transmitter.new(
-          "#{Appsignal.config[:logging_endpoint]}/check_ins/json"
-        )
+        return @transmitter if @transmitter
+
+        NEW_TRANSMITTER_MUTEX.synchronize do
+          @transmitter ||= Transmitter.new(
+            "#{Appsignal.config[:logging_endpoint]}/check_ins/json"
+          )
+        end
+
+        @transmitter
```

Comment:
The transmitter has no state that changes. It doesn't need a mutex necessarily, no?

Better yet, the scheduler should have the transmitter instance. It doesn't need to a global on the CheckIn module. Saves us one mutex too.

Reply:
Yeah, you're right, I was kinda going on auto-pilot here. To be honest, I don't even recall why I made the check-in module have a single shared transmitter in the first place. I'll see if I can get rid of that or if it's ETOOHARD.

Reply:
Super easy, barely an inconvenience. 🚀 

I think I might have done it this way initially because it made it easier to test the check-ins by mocking methods on the transmitter? Which is not how we currently test the transmitter.